bin/check-aggregate-output.rb

```diff
@@ -0,0 +1,209 @@
+         default: 30
+
+  option :check,
+         short: '-c CHECK',
+         long: '--check CHECK',
+         description: 'Aggregate CHECK name',
+         required: true
+
+  option :age,
+         short: '-A SECONDS',
+         long: '--age SECONDS',
+         description: 'Minimum aggregate age in SECONDS, time since check request issued',
+         default: 30,
+         proc: proc(&:to_i)
+
+  option :message,
+         short: '-M MESSAGE',
+         long: '--message MESSAGE',
+         description: 'A custom error MESSAGE'
+
+  option :collect_output,
+         short: '-o',
+         long: '--output',
+         boolean: true,
+         description: 'Collects all non-ok outputs',
+         default: true
+
+  option :warning,
+         short: '-W PERCENT',
+         long: '--warning PERCENT',
+         description: 'PERCENT warning before warning (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :warning_count,
+         long: '--warning_count INTEGER',
+         description: 'number of nodes in warning before warning (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :critical,
+         short: '-C PERCENT',
+         long: '--critical PERCENT',
+         description: 'PERCENT critical before critical (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :critical_count,
+         long: '--critical_count INTEGER',
+         description: 'number of node in critical before critical (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :ignore_severity,
+         long: '--ignore-severity',
+         description: 'Ignore severities, all non-ok will count for critical, critical_count, warning and warning_count option',
+         boolean: true,
+         default: false
+
+  def api_request(resource)
+    verify_mode = OpenSSL::SSL::VERIFY_PEER
+    verify_mode = OpenSSL::SSL::VERIFY_NONE if config[:insecure]
+    request = RestClient::Resource.new(config[:api] + resource, timeout: config[:timeout],
+                                                                user: config[:user],
+                                                                password: config[:password],
+                                                                verify_ssl: verify_mode)
+    JSON.parse(request.get)
+  rescue Errno::ECONNREFUSED
+    warning 'Connection refused'
+  rescue RestClient::RequestFailed
+    warning 'Request failed'
+  rescue RestClient::RequestTimeout
+    warning 'Connection timed out'
+  rescue RestClient::Unauthorized
+    warning 'Missing or incorrect Sensu API credentials'
+  rescue JSON::ParserError
+    warning 'Sensu API returned invalid JSON'
+  end
+
+  def aggregate_results
+    results = api_request("/aggregates/#{config[:check]}?max_age=#{config[:age]}")['results']
+    warning "No aggregates found in last #{config[:age]} seconds" if %w(ok warning critical unknown).all? { |x| results[x].zero? }
```

Comment:
might make more sense to leverage something like [this](https://github.com/sensu-plugins/sensu-plugin/blob/v2.1.0/lib/sensu-plugin/cli.rb#L26-L32) for future proofing new severities. While this is unlikely to occur in the short term as sensu has adhered to the check spec of nagios with sensu 2.x on the horizon it's possible they might want to enhance on top of the nagios check spec.